### ISSUE TITLE
No user scopes leads to error

### ISSUE THREAD

**Issue by jennyf19** [AzureAD/microsoft-identity-web#14]
from @onovotny and copied from https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/issues/241

If you want to use .AddMsal(), there's no way to do so without any user scopes. It might be needed for Application flows.

If you use `new string[0]` for the scopes parameter, it'll throw in the OnAuthorizationCodeReceived callback since options.Scope will be null:
`await tokenAcquisition.AddAccountToCacheFromAuthorizationCodeAsync(context, options.Scope).ConfigureAwait(false);`

**Comment by bgavrilMS** [comment]
Not sure this is a work item for the extenion. MSAL needs to support this scenario, and it is not trivial (but not super complex either). Tracking work item: 

https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/715

**Comment by jmprieur** [comment]
@jennyf19 : proposing to take it in 0.2 given that MSAL.NET fixed it;

**Comment by bgavrilMS** [comment]
This should work now and the behaviour of passing no scopes I observed is as follows: 

- MSAL will still pass in the required scopes for itself to function - `offline_access`, `openid`, `profile`
- AAD and ADFS do issue an access token (I think this wasn't the case some time ago). I assume it has to do with the fact that profile is transformed to User.Read.
- B2C does not issue an access token. 
- IDTokens are always issued
- `AcquireTokenSilent` with no scopes is not supported on B2C.

**Comment by jmprieur** [comment]
Spec: @jennyf19 @pmaytak 

**Why?**
It should be possible to write:

```CSharp
.AddSignIn(Configuration)
.AddWebAppCallsProtectedWebApi(Configuration)
.AddInMemoryTokenCaches();
```

and then have the scopes requested in the Web app controllers for a complelety dynamic experience (with incremental consent, or with pre-authorized scopes)

**What?**
The proposal (non breaking) is to enable initialScopes to be null or an empty string[] in `AddWebAppCallsProtectedWebApi`.
https://github.com/AzureAD/microsoft-identity-web/blob/4cfedc9771ff75ec611f48f504dd0eea546e1cdc/src/Microsoft.Identity.Web/WebAppServiceCollectionExtensions.cs#L104-L109

**Comment by pmaytak** [comment]
Proposal makes sense.

**Comment by jennyf19** [comment]
Included in [0.1.3-preview release](https://github.com/AzureAD/microsoft-identity-web/releases/tag/0.1.3-preview)
